### DISCUSSION TITLE
Cannot setup Prisma Logging.

### DISCUSSION THREAD

**Discussion by Jeffr-K** [prisma/prisma#14754] [Q&A] [answered]
I want to see the query log on terminal. I am using nestjs, prisma version 2.
How do I configure for watching the query logs on terminal whenever getting http request??

```
import { Injectable, OnModuleInit, OnModuleDestroy, INestApplication, Logger } from "@nestjs/common";
import { PrismaClient } from '@prisma/client';
import { config } from 'dotenv';
config();

@Injectable()
export class PrismaService
  extends PrismaClient
  implements OnModuleInit, OnModuleDestroy {
  private readonly logger = new Logger(PrismaService.name);

  constructor() {
    super({
      log: [
        { emit: 'event', level: 'query' },
        { emit: 'stdout', level: 'info' },
        { emit: 'stdout', level: 'warn' },
        { emit: 'stdout', level: 'error' },
      ],
    });
  }

  async onModuleInit() {
    await this.$connect();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    this.$on('query', async (e) => {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      console.log(`${e.query} ${e.params}`);
    });
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    this.$on('error', (event) => {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      this.logger.verbose(event.target);
    });
  }

  async onModuleDestroy() {
    await this.$disconnect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.$on('beforeExit', async (event) => {
      console.log(event.name);
      await app.close();
    });
  }
}
```

**Comment by nurul3101** [comment, marked as answer]
Hi @Jeffr-K 👋 
Did you have a look at this [answer](https://www.anycodings.com/questions/logging-with-prisma-2-and-nestjs-dependency-injection-problem)? They seemed to have the same problem and the solution mentioned in the link seems to work. Can you try it?

**Reply by Jeffr-K** [reply]
@nurul3101 Thank you for response about my question :> 
I already found a solution about it through your recommend coment. 
I appreciate your hard working :)

**Reply by nurul3101** [reply]
I am glad to hear that! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
